### PR TITLE
[CALCITE-3177]Ensure correct deserialization of relational algebra.

### PR DESCRIPTION
**Pull request by yanlin-Lynn**
When deserialize from json string, in some cases, different SqlOperators may have the same name, for example, "PLUS" and "PLUS_PREFIX". So the kind of SqlOperator should also be used for searching.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/plan/RelWriterTest.java [hunk 1]
```diff
@@ -472,6 +508,40 @@
...
             + "    LogicalTableScan(table=[[hr, emps]])\n"));
   }
+
+  @Test public void testPlusOperator() {
+    String sql = "select SAL + 10 FROM SALES.EMP";
+    RelRoot relRoot = tester.convertSqlToRel(sql);
+    RelJsonWriter jsonWriter = new RelJsonWriter();
+    relRoot.rel.explain(jsonWriter);
+    String relJson = jsonWriter.asString();
+    System.out.println(relJson);
```
Comment (michaelmior) [inline core/src/test/java/org/apache/calcite/plan/RelWriterTest.java]: This print statement should be removed.

Comment (yanlin-Lynn) [reply core/src/test/java/org/apache/calcite/plan/RelWriterTest.java]: OK. Thanks for reviewing.

**Comment by michaelmior** [comment]
One final thing is that the commit message should reflect what you fixed. Reading the current commit message out of context doesn't really indicate what the change is for. Something like "Ensure correct deserialization of relational algebra" or similar would be more helpful.

**Comment by yanlin-Lynn** [comment]
> One final thing is that the commit message should reflect what you fixed. Reading the current commit message out of context doesn't really indicate what the change is for. Something like "Ensure correct deserialization of relational algebra" or similar would be more helpful.

OK.

**Pull request merged**